XenAdmin/Wizards/NewSRWizard_Pages/Frontends/VHDoNFSNew.cs

```diff
@@ -0,0 +1,372 @@
+            NfsScanButton.Enabled = SrWizardHelpers.ValidateSharePath(NfsSharePathTextBox.Text);
+            serverScanButton.Enabled = SrWizardHelpers.ValidateServerAddress(ServerTextBox.Text);
+            OnPageUpdated();
+        }
+
+        private void ServerTextBox_TextChanged(object sender, EventArgs e)
+        {
+            serverScanButton.Enabled = SrWizardHelpers.ValidateServerAddress(ServerTextBox.Text);
+            ToggleReattachControlsEnabledState(false);
+            UpdateButtons();
+        }
+
+        private void NfsServerPathTextBox_TextChanged(object sender, EventArgs e)
+        {
+            NfsScanButton.Enabled = SrWizardHelpers.ValidateSharePath(NfsSharePathTextBox.Text);
+
+            listBoxNfsSRs.Items.Clear();
+            ToggleReattachControlsEnabledState(false);
+
+            if(radioButtonNfsNew.Enabled)
+                radioButtonNfsNew.Checked = true;
+
+            nfsVersion3RadioButton.Enabled = nfsVersion4RadioButton.Enabled = true;
+
+            UpdateButtons();
+        }
+
+        private void radioButtonNfsReattach_CheckedChanged(object sender, EventArgs e)
+        {
+            radioButtonNfsNew.Checked = !radioButtonNfsReattach.Checked;
+            UpdateButtons();
+        }
+
+        private void radioButtonNfsNew_CheckedChanged(object sender, EventArgs e)
+        {
+            if (radioButtonNfsNew.Checked)
+                listBoxNfsSRs.SelectedIndex = -1;
+            
+            radioButtonNfsReattach.Checked = !radioButtonNfsNew.Checked;
+            UpdateButtons();
+        }
+
+        private void serverScanButton_Click(object sender, EventArgs e)
+        {
+            serverScanButton.Enabled = false;
+            Dictionary<String, String> dconf = new Dictionary<String, String>();
+            dconf[SERVER] = ServerTextBox.Text;
+
+            Host master = Helpers.GetMaster(Connection);
+            if (master == null)
+                return;
+            
+            SrProbeAction action = new SrProbeAction(Connection, master, SR.SRTypes.nfs, dconf);
+            using (var dialog = new ActionProgressDialog(action, ProgressBarStyle.Marquee))
+            {
+                dialog.ShowCancel = true;
+                dialog.ShowDialog(this);
+            }
+
+            try
+            {
+                if (String.IsNullOrEmpty(SrProbeAction.XMLFromErrorDescription))
+                {
+                    DisableControls();
+                    return;
+                }
+                                
+                NfsSharePathTextBox.Enabled = serverOptionsTextBox.Enabled = serverScanButton.Enabled = sharePathExampleLabel.Visible = true;
+                GetSupportedNfsVersionsAndSetUI(SrProbeAction.XMLFromErrorDescription);
+
+                //clear xml string otherwise it will persist across different probes and be reused if new one is not created, bypassing above if statement
+                SrProbeAction.XMLFromErrorDescription = null;

```

Comment:
I can't see why we have a try-finally block here, but I'm fine with it.
However, if we have it around, this line is a typical example that should be in the finally block - to make sure it is always executed.
Therefore, I suggest we move this into the finally.